app/models/miq_report/seeding.rb

```diff
@@ -7,14 +7,18 @@ module MiqReport::Seeding
   module ClassMethods
     def seed
       transaction do
+        # seeding from files, :filename attribute of existing record may be changed in this process
+        seed_files.each do |f|
+          seed_record(f, MiqReport.find_by(:filename => seed_filename(f)))
```

Comment:
Find by in a loop is bad on performance (N+1), you should be able to do this by just querying the whole table and index_by(&:filename)

Reply:
I'm also concerned that the query doesn't handle rpt_type here like it does before, but I think I need a clearer head to understand that better... Let's chat tomorrow